src/us/kbase/workspace/database/provenance/Title.java

```diff
@@ -0,0 +1,171 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Optional;
+import us.kbase.workspace.database.Util;
+
+/**
+ * Represents a title or name of a resource.
+ */
+public class Title {
+
+	private final String titleString;
+	private final TitleType titleType;
+	private final String titleLanguage;
+
+	/**
+	 * An enum representing types of title.
+	 */
+	public enum TitleType {
+		SUBTITLE,
+		ALTERNATIVE_TITLE,
+		TRANSLATED_TITLE,
+		OTHER;
+
+		private static final Map<String, TitleType> STRING_TO_TITLE_TYPE_MAP = new HashMap<>();
+		static {
+			for (final TitleType tt: TitleType.values()) {
+				STRING_TO_TITLE_TYPE_MAP.put(tt.name().toLowerCase(), tt);
+			}
+		}
+
+		/** Gets a title type based on a supplied string.
+		 * @param input a string representing a title type.
+		 * @return a TitleType.
+		 * @throws IllegalArgumentException if there is no title type related to the input string.
+		 */
+		public static TitleType getTitleType(final String input) {
+			final String lowercaseInput = Util.checkString(input, "input").toLowerCase();
+			if (STRING_TO_TITLE_TYPE_MAP.containsKey(lowercaseInput)) {
+				return STRING_TO_TITLE_TYPE_MAP.get(lowercaseInput);
+			}
+			throw new IllegalArgumentException("Invalid titleType: " + input);
+		}
+	}
+
+	private Title(
+			final String titleString,
+			final TitleType titleType,
+			final String titleLanguage) {
+		this.titleString = titleString;
+		this.titleLanguage = titleLanguage;
+		this.titleType = titleType;
+	}
+
+	/**
+	 * Gets the title of the resource.
+	 *
+	 * @return the titleString.
+	 */
+	public String getTitleString() {
+		return titleString;
+	}
+
+	/**
+	 * Gets the type of title being represented.
+	 *
+	 * @return the titleType, if present.
+	 */
+	public Optional<TitleType> getTitleType() {
+		return Optional.ofNullable(titleType);
+	}
+
+	/**
+	 * Gets the language in which the title is written.
+	 *
+	 * @return the titleLanguage, if present.
+	 */
+	public Optional<String> getTitleLanguage() {
```

Comment:
I feel like this shouldn't be optional but I don't understand the context

Reply:
With both this and the titleType, I was weighing up whether to have default values (`en` in this case, plain old title in the case of `titleType`) or whether to leave them null if the value is unknown or unrecorded. I went for the conservative approach, leaving them null by default, as there's less likelihood that a title would have an incorrect titleType or titleLanguage, than if defaults were assigned. This is what DataCite, from whom we will be sourcing a fair bit of our data, do.

Reply:
Fair enough.